src/roman_datamodels/datamodels.py

```diff
@@ -300,6 +303,57 @@ class GuidewindowModel(DataModel):
     pass
 
 
+class ModelContainerModel(DataModel, Sequence):
+    # Needs a number of methods to properly handle the contents
+    def __init__(self, init=None, asn_schema= None, asn_file_path=None, model_file_path=None, iscopy=False, **kwargs):
+        # __init__(self, init=None, asn_exptypes=None, asn_n_members=None,
+        #                  iscopy=False, **kwargs):
+        super().__init__(init=None, **kwargs)
+
+        self._models = []
+        self._iscopy = iscopy
+        # self.asn_exptypes = asn_exptypes
+        # self.asn_n_members = asn_n_members
+        # self.asn_table = {}
+        # self.asn_table_name = None
+        # self.asn_pool_name = None
+
+        self._memmap = kwargs.get("memmap", False)
+        self._return_open = kwargs.get('return_open', True)
+        self._save_open = kwargs.get('save_open', True)
+
+        if asn_file_path:
+            with builtins.open(asn_file_path, 'r') as asn_file:
+                asn_schema = json.load(asn_file)
+
+        for product in asn_schema['products']:
+            for member in product['members']:
+                if self._iscopy:
+                    member_model = open(model_file_path + member['expname'])
+                else:
+                    member_model = "null"
```

Comment:
Could you add a comment here why the above logic is necessary? It's not obvious to me.
Also it's worth investigating the use of `_iscopy` in jwst and whether it's needed for roman.

Reply:
Also this section of the code assumes the input is an association table and that asn_schema is created in the previous code block or passed as an argument in the init. However, if the input is a list of models, or both `asn_schema` and `asn_file_path` are None  this will crash.

Reply:
And it does not even seem to be working with an association,

```
>>> a2 = builtins.open('r00001-o001_image_001_asn.json')
>>> c1 = rdm.datamodels.ModelContainerModel(a2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ddavis/miniconda3/envs/rcal_dev/lib/python3.10/site-packages/roman_datamodels/datamodels.py", line 329, in __init__
    for product in asn_schema['products']:
TypeError: 'NoneType' object is not subscriptable
```